protocol/nats/v2/subscriber.go

```diff
@@ -1,20 +1,27 @@
 package nats
 
 import (
+	"github.com/nats-io/jsm.go"
 	"github.com/nats-io/nats.go"
+	"time"
 )
 
 // The Subscriber interface allows us to configure how the subscription is created
 type Subscriber interface {
-	Subscribe(conn *nats.Conn, subject string, cb nats.MsgHandler) (*nats.Subscription, error)
+	Subscribe(conn *nats.Conn, subject string, cb nats.MsgHandler) (Subscription, error)
```

Comment:
This change will require a v3.0.0 semver bump, and break the downstream integrators.

Can you think of a way to do this change without changing the API? Do we need to introduce a new protocol type or option for NATS?